### PR TITLE
Add reference commands

### PR DESCRIPTION
**Pull request by polandll**
Reference CQL commands that will be referenced in the CEP-30 Vector Search PR. I did not create a ticket for this work.

Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

File: doc/modules/cassandra/pages/reference/index.adoc [hunk 1]
```diff
@@ -1,4 +1,3 @@
 = Reference
 
-* xref:cassandra:reference/cql-commands/commands-toc.adoc[CQL commands]
-* xref:cassandra:reference/java17.adoc[Java 17 usage]
\ No newline at end of file
+* xref:reference/cql-commands/commands-toc.adoc[CQL commands]
```
Comment (michaelsembwever) [inline doc/modules/cassandra/pages/reference/index.adoc:3]: intended to remove java17.adoc link ?

File: doc/modules/cassandra/pages/reference/cql-commands/commands-toc.adoc
Comment (michaelsembwever) [inline doc/modules/cassandra/pages/reference/cql-commands/commands-toc.adoc:1]: lots of commented out xref lines in this file, is that intended ?

File: doc/modules/cassandra/pages/reference/index.adoc (hunk 1 above)
Comment (polandll) [reply doc/modules/cassandra/pages/reference/index.adoc:3]: I did not intend to delete java17.adoc link.

File: doc/modules/cassandra/pages/reference/cql-commands/commands-toc.adoc
Comment (polandll) [reply doc/modules/cassandra/pages/reference/cql-commands/commands-toc.adoc:1]: Yes, they will be added over time.

**Pull request closed without merging**